src/rviz/default_plugin/marker_utils.cpp

```diff
@@ -76,4 +84,346 @@ MarkerBase* createMarker(int marker_type, MarkerDisplay* owner, DisplayContext*
   }
 }
 
+bool checkMarkerMsg(const visualization_msgs::Marker& marker, MarkerDisplay* owner)
+{
+
+  if(marker.action != visualization_msgs::Marker::ADD)
+    return true;
+
+  std::stringstream ss;
+  switch (marker.type) {
+  case visualization_msgs::Marker::ARROW:
+    checkQuaternion(marker, ss);
+    checkScale(marker, ss);
+    checkColor(marker, ss);
+    checkPointsArrow(marker, ss);
+    checkColorsEmpty(marker, ss);
+    checkTextEmpty(marker, ss);
+    checkMeshEmpty(marker, ss);
+    break;
+
+  case visualization_msgs::Marker::CUBE:
+  case visualization_msgs::Marker::CYLINDER:
+  case visualization_msgs::Marker::SPHERE:
+    checkQuaternion(marker, ss);
+    checkScale(marker, ss);
+    checkColor(marker, ss);
+    checkPointsEmpty(marker, ss);
+    checkColorsEmpty(marker, ss);
+    checkTextEmpty(marker, ss);
+    checkMeshEmpty(marker, ss);
+    break;
+
+
+  case visualization_msgs::Marker::LINE_STRIP:
+  case visualization_msgs::Marker::LINE_LIST:
+    checkQuaternion(marker, ss);
+    checkScaleLineStripAndList(marker, ss);
+    checkPointsNotEmpty(marker, ss);
+    checkColors(marker, ss);
+    checkTextEmpty(marker, ss);
+    checkMeshEmpty(marker, ss);
+    break;
+
+  case visualization_msgs::Marker::SPHERE_LIST:
+  case visualization_msgs::Marker::CUBE_LIST:
+  case visualization_msgs::Marker::TRIANGLE_LIST:
+    checkQuaternion(marker, ss);
+    checkScale(marker, ss);
+    checkPointsNotEmpty(marker, ss);
+    checkColors(marker, ss);
+    checkTextEmpty(marker, ss);
+    checkMeshEmpty(marker, ss);
+    break;
+
+  case visualization_msgs::Marker::POINTS:
+    checkScalePoints(marker, ss);
+    checkPointsNotEmpty(marker, ss);
+    checkColors(marker, ss);
+    checkTextEmpty(marker, ss);
+    checkMeshEmpty(marker, ss);
+    break;
+
+  case visualization_msgs::Marker::TEXT_VIEW_FACING:
+    checkColor(marker, ss);
+    checkScaleText(marker, ss);
+    checkTextNotEmptyOrWhitespace(marker, ss);
+    checkPointsEmpty(marker, ss);
+    checkColorsEmpty(marker, ss);
+    checkMeshEmpty(marker, ss);
+    break;
+
+  case visualization_msgs::Marker::MESH_RESOURCE:
+    checkQuaternion(marker, ss);
+    checkColor(marker, ss);
+    checkScale(marker, ss);
+    checkPointsEmpty(marker, ss);
+    checkColorsEmpty(marker, ss);
+    checkTextEmpty(marker, ss);
+    break;
+
+  default:
+    ss << "Unknown marker type: " << marker.type ;
+  }
+
+  if(ss.tellp() != 0) //stringstream is not empty
+  {
+    std::string warning = ss.str();
+
+    owner->setMarkerStatus(MarkerID(marker.ns, marker.id), StatusProperty::Warn, warning);
+    ROS_WARN("Marker '%s/%d': %s", marker.ns.c_str(), marker.id, warning.c_str());
```

Comment:
All messages have now the same severity level of "warning". Previously, e.g. an unknown marker type would raise an error, quaternion issues would raise a warning only _once_.
I suggest to add another reference argument to your check functions and pass the desired warning level.
Initialized to Ok, the checkers can further increase the warning level (but not decrease).

Reply:
Firstly, do you think that the warnings should only be printed once as it was before?

Secondly, I agree that for some messages ROS_INFO would be enough. The others should be ROS_WARN when following the definition [here](http://wiki.ros.org/Verbosity%20Levels). I do not see a need for ROS_ERROR. What is your opinion on this?

Reply:
> do you think that the warnings should only be printed once as it was before?

ROS_*_ONCE is not an option here because the notification can be triggered by any error type.
To properly reduce warnings in a stream of repeating messages, the display would have to maintain a memory of which namespace / error combination was observed before.
The user (or at least I) would expect at least one notification per "type" of marker, not just one per display.
I believe the required overhead outweighs the benefits here.
> for some messages ROS_INFO would be enough. The others should be ROS_WARN when following the definition [here](http://wiki.ros.org/Verbosity%20Levels). I do not see a need for ROS_ERROR. What is your opinion on this?

Firstly, this is not only about `ROS_*`, rviz also defines their own status family to display in the GUI (in the line right above), which is similar though.

Secondly, if a vital part of the marker message is filled in incorrectly and the marker cannot be displayed due to that,
this constitutes an actual error, not just a warning.
I have a hard time coming up with a message that could constitute an `Info`-style notification, but it might exist.

Reply:
Please excuse the delay on this, I finally got to implementing this. All issues which cause the marker not to be displayed are now an error. All issues that are related to additional information in the marker which is not used in this type of marker are a warning. 3aadbff